### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/rr/actions/runs/4276402800#summary-11608769511

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/rr/actions/runs/4276403008#summary-11608769777

---

- [ ] Did you run the rr test suite (including your new tests, if any), and pass all the tests?  `make -C $objdir check`.  Unfortunately, rr doesn't have automated infrastructure that can run the tests yet, so developers have to run them locally.

  -- Unfortunately, I'm running on macOS, and I generally rely on CI...I tried running checks in GitHub and they appeared to take >6hrs...

- [x] Does your PR apply cleanly on top of upstream/master HEAD?  It's dangerous to have someone else sort out your merge conflicts, so just don't do it.  Best of all is to have a PR *rebased* on top of upstream/master HEAD, so that the merge is simply a fast-forward.

- [x] If your PR includes multiple changesets, do they all (i) build cleanly in sequence;
- [ ]  (ii) pass all tests in sequence?  This is important for bisecting over commit history.

- [x] If your PR is large or includes many changesets, would it have been possible to break the changes into a series of smaller PRs?  For example, it's hard to review a big patch that, say, fixes whitespace errors in a file along with a one-line, important, bug fix.  It's much easier to review one PR that fixes whitespace (which can just be skimmed), and then review another PR that makes the one-line bug fix (which would be scrutinized more).  This approach is also better for the patch author in that it usually allows the work to land faster, and reduces the burden of continually un-bit-rotting large, trivial, changes.

  -- It's possible to split almost any of these commits into their own PRs, but I'd rather let the project decide.

- [x] Did you check your code is formatted correctly? It's easiest to run `scripts/reformat.sh` on each commit.

### CODE REVIEW

File: src/RecordTask.h [hunk 1]
```diff
@@ -584,7 +584,7 @@ class RecordTask final : public Task {
   bool is_container_init() const { return tg->tgid_own_namespace == 1; }
 
   /**
-   * Linux requires the invariant that that all members of a thread group
```
Comment (jsoref) [inline src/RecordTask.h:587]: sometimes doubled `that` is intentional. It didn't appear to be in this repository

File: src/test/x86/ptrace_tls.c [hunk 2]
```diff
@@ -118,7 +118,7 @@ int main(void) {
                           parent_desc.entry_number, &parent_desc));
   test_assert(parent_desc.limit == 2);
 
-  /* Next we're going to verify that these are actually reflected in to the LDT
+  /* Next we're going to verify that these are actually reflected into the LDT
```
Comment (jsoref) [inline src/test/x86/ptrace_tls.c:121]: `in to`/`into` changes have been controversial. I'm happy to drop this or any other change

**Comment by rocallahan** [comment]
Hello Timeless!

File: third-party/brotli/enc/hash_composite_inc.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ static void FN(Initialize)(
   self->ha = 0;
   self->hb = 0;
   self->params = params;
-  /* TODO: Initialize of the hashers is defered to Prepare (and params
+  /* TODO: Initialize of the hashers is deferred to Prepare (and params
```
Comment (rocallahan) [inline third-party/brotli/enc/hash_composite_inc.h:46]: Don't patch anything in third-party. These sources are imported unchanged from upstream sources so any local changes just make our lives difficult.

**Review by rocallahan — CHANGES_REQUESTED** [review]

**Comment by khuey** [comment]
I don't have any strong opinions about the patches but I do want these commits to be squashed when this is merged.

**Comment by rocallahan** [comment]
Yes, please squash now.

**Comment by jsoref** [comment]
Do you have any preference about the commit message?

GitHub can let repository owners squash during merge.

(Otherwise, I'll be able to squash in a few hours.)

**Comment by khuey** [comment]
Yeah we can squash it on merge.

**Comment by rocallahan** [comment]
Thanks!

**Pull request merged**